mtgjson5/providers/tcgplayer.py

```diff
@@ -147,6 +147,124 @@ def generate_today_price_dict(
+        :param all_printings_path Path to AllPrintings.json for pre-processing
+        :return: Buylist Prices to combine, not in correct format yet
+        """
+        if not self.__keys_found:
+            LOGGER.warning("Keys not found for TCGPlayer, skipping")
+            return []
+        uuid_map = generate_tcgplayer_to_mtgjson_map(all_printings_path)
+        CACHE_PATH.mkdir(parents=True, exist_ok=True)
+        with CACHE_PATH.joinpath("tcgplayer_buylist_price_map.json").open("w") as file:
+            json.dump(uuid_map, file)
+
+        ids_and_names = self.get_tcgplayer_magic_set_ids()
+        set_price_data = [
+            get_tcgplayer_buylist_prices_map(set_ids) for set_ids in ids_and_names
+        ]
+
+        return set_price_data
+
+
+def get_tcgplayer_sku_data(group_id_and_name: Tuple[str, str]) -> List[Dict[str, Any]]:
+    """
+    Finds all sku data for a given group using the TCGPlayer API
+    :param group_id_and_name: group id and name for the set to get data for
+    :return: product data including skus to be parsed into a sku map
+    """
+    magic_set_product_data = []
+    api_offset = 0
+
+    while True:
+        api_response = TCGPlayerProvider().download(
+            "https://api.tcgplayer.com/catalog/products",
+            {
+                "offset": str(api_offset),
+                "limit": 100,
+                "categoryId": 1,
+                "includeSkus": True,
+                "groupId": group_id_and_name[0],
+            },
+        )
+
+        if not api_response:
+            # No more entries
+            break
+
+        response = json.loads(api_response)
+        if not response["results"]:
+            # Something went wrong
+            break
+
+        magic_set_product_data.extend(response["results"])
+
+        api_offset += len(response["results"])
+
+    return magic_set_product_data
+
+
+def generate_tcgplayer_sku_map(
+    tcgplayer_set_sku_data: List[Dict[str, Any]],
+) -> Dict[str, Dict[str, Optional[int]]]:
+    """
+    takes product info and builds a sku map
+    :param tcgplayer_set_sku_data: list of product data dicts used to a build a product id to sku map
+    :return: Map of TCGPlayerID -> NM Foil and Nonfoil SKU
+    """
+    tcgplayer_sku_map: Dict[str, Dict[str, Optional[int]]] = {}
+    for product_data in tcgplayer_set_sku_data:
+        nonfoil_sku: Optional[int] = None
+        foil_sku: Optional[int] = None
+        for sku in product_data["skus"]:
+            if (
+                sku["conditionId"] == 1
+                and sku["printingId"] == 1
+                and sku["languageId"] == 1
+            ):
+                nonfoil_sku = sku["skuId"]
+            elif (
+                sku["conditionId"] == 1
+                and sku["printingId"] == 2
+                and sku["languageId"] == 1
+            ):
+                foil_sku = sku["skuId"]
+        tcgplayer_sku_map[str(product_data["productId"])] = {
```

Comment:
I'd honestly be inclined to not include NULL values into this dict, and use the .get() method downstream... would remove some complexity in types